docs/mp/testing/02_testing_ng.adoc

```diff
@@ -0,0 +1,107 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Testing with Test NG
+:h1Prefix: MP
+:pagename: testing_ng
+:description: Helidon Testing with TestNG
+:keywords: helidon, mp, test, testing, testng
+:feature-name: Testing with TestNG
+:common-deps-page-prefix-inc: ../../shared/dependencies/common_shared.adoc
+
+Helidon provides built-in test support for CDI testing in TestNG.
+
+include::{common-deps-page-prefix-inc}[tag=maven-dependency]
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.microprofile.tests</groupId>
+    <artifactId>helidon-microprofile-tests-testng</artifactId>
+    <scope>test</scope>
+</dependency>
+----
+
+== Usage - default
+A test can be annotated with `io.helidon.microprofile.tests.testng.HelidonTest` annotation to mark it as a
+CDI test. This annotation will start the CDI container before any test method is invoked, and stop it after
+the last method is invoked. This annotation also enables injection into the test class itself.
+
+The annotations described in this section are inherited (for the non-repeatable ones), and additive (for repeatable).
+So if you declare `@DisableDiscovery` on abstract class, all implementations will have discovery disabled, unless you
+annotate the implementation class with `@DisableDiscovery(false)`.
+If you declare `@AddBean` on both abstract class and implementation class, both beans will be added.
+
+In addition to this simplification, the following annotations are supported:
+
+- `io.helidon.microprofile.tests.testng.AddBean` - to add one or more beans to the container
+        (if not part of a bean archive, or when discovery is disabled)
+- `io.helidon.microprofile.tests.testng.AddExtension` - to add one or more CDI extensions to the container
+        (if not added through service loader, or when discovery is disabled)
+- `io.helidon.microprofile.tests.testng.AddConfig` - to add one or more configuration properties to MicroProfile config
+        without the need of creating a `microprofile-config.properties` file
+- `io.helidon.microprofile.tests.testng.DisableDiscovery` - to disable automated discovery of beans and extensions
+
+[source,java]
+.Code sample
+----
+@HelidonTest
+@DisableDiscovery
+@AddBean(MyBean.class)
+@AddExtension(ConfigCdiExtension.class)
+@AddConfig(key = "app.greeting", value = "TestHello")
+class TestNoDiscovery {
+    @Inject
+    private MyBean myBean;
+
+    @Test
+    void testGreeting() {
+        assertThat(myBean, notNullValue());
+        assertThat(myBean.greeting(), is("TestHello"));
+    }
+}
+----
+
+== Usage - per method CDI container
+A test can be annotated as follows:
+
+`@HelidonTest(resetPerTest = true)`
```

Comment:
I think this should not be allowed at all (and the test should fail with message saying `resetPerTest` is not supported).
This whole section will not work as described, and is not much useful, when you cannot obtain a client to use.

Reply:
Interestingly, that I had some support requests from users, that require just to restart the container on each test. With no changes to configuration. I would rather keep this option for now.